src/tests/ftest/telemetry/telemetry_pool_metrics.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import re
+
+from avocado.core.exceptions import TestFail
+from ior_test_base import IorTestBase
+from telemetry_test_base import TestWithTelemetry
+
+
+class TelemetryPoolMetrics(IorTestBase, TestWithTelemetry):
+    # pylint: disable=too-many-ancestors
+    # pylint: disable=too-many-nested-blocks
+    """Test telemetry pool basic metrics.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a TelemetryPoolMetrics object."""
+        super().__init__(*args, **kwargs)
+        # remove this variable and it's use from the code,
+        # once DAOS-8592 is resolved.
+        self.threshold_percent = None
+
+    def check_range(self, expected_value, actual_value, threshold_percent):
+        """Check if actual value is within expected
+           range of expected value
+        Args:
+          expected_value: Provide expected value to be checked against
+          actual_value: Provide the actual value to be checked
+          threshold_percent: Threshold percentage for acceptable range
+
+        Return:
+          bool: Returns boolean value depending of the check. Returns
+                True if the value is within range, else False.
+
+        """
+        # Calculate the lower and upper bound values
+        upper_bound = (expected_value +
+                       (threshold_percent / 100 * expected_value))
+        lower_bound = (expected_value -
+                       (threshold_percent / 100 * expected_value))
+        self.log.debug("Lower Bound: %s", lower_bound)
+        self.log.debug("Actual Value: %s", actual_value)
+        self.log.debug("Upper Bound: %s", upper_bound)
+        # Check if the actual value is within range
+        if lower_bound <= actual_value <= upper_bound:
+            return True
+
+        return False
+
+    def verify_pool_metrics(self, pool_metrics_list, metrics_data,
+                            expected_values):
+        """ Verify telemetry pool metrics from metrics_data.
+
+        Args:
+            pool_metrics_list (list): list of telemetry pool metrics.
+            metrics_data (dict): a dictionary of host keys linked to a
+                                 list of pool metric names. Contains
+                                 data before and after read/write.
+            expected_values (list): List of expected values for different
+                                    pool metrics.
+
+        """
+        # initializing method params
+        total = []
+        final_value = []
+        # collecting data to be verified
+        for name in sorted(pool_metrics_list):
+            self.log.debug("  --telemetry metric: %s", name)
+            for key in sorted(metrics_data):
+                m_data = metrics_data[key]
+                total.append(0)
+                total[key] = 0
```

Comment:
I guess the dictionary keys are ints? Zero for before, one for after? Intermixing dict keys with list indices is a little confusing.

Reply:
Thanks for pointing. I have made the change and instead using a temporary variable now.